### ISSUE TITLE
Before specific feature

### ISSUE THREAD

**Issue by remcowesterhoud** [behave/behave#456]
As far as I'm aware there is at this moment only an option to have a before_feature hook. This will run before every feature I have. To prevent myself from getting a single before_feature containing dozens of if statements for each and every feature I have, I was wondering if there is a good way to have a feature specific before that is only ran for a single feature (like the setup_class form pytest).

I've been trying to do it using the background so I can easily define and reuse the step, with different settings. But it executed before every scenario and not each feature. Is there a way to make a step execute only once per feature?
Something like a Feature Background would be perfect.

**Comment by gregsaun** [comment]
You can create a scenario with the tag "@setup" like here: https://github.com/behave/behave/blob/master/features/formatter.tags.feature

The problem is that if you have to run the feature by giving the path to the .feature file (`behave features/myfeat.feature`) and not the tag (`behave -t @slow`). Would be nice to have a setup and teardown functions called when using tags too.

**Comment by jenisys** [comment]
@remcowesterhoud 
If you want to simplify a specific setup, like a fixture in [py.test](http://pytest.org/latest/fixture.html#fixture), on a feature, it may to be good idea to use ``tags`` to mark features/scenarios that require a specific setup/teardown logic (basically a fixture). These tags should then not used for feature/scenario selection
because they serve a different purpose. They are used as markers to identify setup/teardown strategies (on for each fixture) in your hook (like ``before_feature()``, ``before_tag()``, ...).

Note that use steps in the environment is not a good idea (IMHO). Your steps are based on programming logic. Therefore, it should be easy for you to make direct use of that instead of using the steps.
If you have a hammer, not everything is a nail. There are also screws (and yes I know that you can hammer them also into a wall) and more...

**Comment by remcowesterhoud** [comment]
@jenisys 
Cool, thanks for clearing this up for me.
